### PR TITLE
Fix xml literal issues

### LINKED ISSUE #23938: XML template literal fails in transformer
**Description:**

Issue is here: https://github.com/ballerina-platform/ballerina-lang/blob/fb4c8f31b6150f6ee3ab50e0c76176eaa015a1cd/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java#L1723

it only consider the first child.

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
$subject

Fixes #23938

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLLiteralTest.java [hunk 1]
```diff
@@ -156,13 +155,13 @@ public void testXMLPILiteral() {
         Assert.assertEquals(returns[1].stringValue(), "<?foo 11?>");
 
         Assert.assertTrue(returns[2] instanceof BXML);
-        Assert.assertEquals(returns[2].stringValue(), "<?foo  aaa11bbb22ccc?>");
```
Comment (manuranga) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLLiteralTest.java:159]: why are the test changed?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java [hunk 2]
```diff
@@ -1805,7 +1805,16 @@ public BLangNode transform(TemplateExpressionNode expressionNode) {
...
         switch (kind) {
             case XML_TEMPLATE_EXPRESSION:
-                return expressionNode.content().get(0).apply(this);
+                SyntaxKind contentKind = expressionNode.content().get(0).kind();
+                switch (contentKind) {
+                    case XML_COMMENT:
+                    case XML_PI:
+                    case XML_ELEMENT:
+                    case XML_EMPTY_ELEMENT:
+                        return expressionNode.content().get(0).apply(this);
```
Comment (SupunS) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:1814]: shouldn't we use createExpression() method?

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLLiteralTest.java (hunk 1 above)
Comment (SupunS) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLLiteralTest.java:159]: This is a bug in the old xml-parser, where ws is captured as part of the xml-pi `char*` section. This is fixed in the new parser, now any ws is capture as ws.

Now we are hitting a different bug which is there in BLangNodes, where ws is not propagated to runtime - BLandNodes ignore ws when creating XMLs. @KavinduZoysa shall we create and issue for this?

Comment (SupunS) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLLiteralTest.java:159]: This is a bug in the old xml-parser, where ws is captured as part of the xml-pi char* section. This is fixed in the new parser, now any ws is capture as ws.

Now we are hitting a different bug which is there in BLangNodes, where ws is not propagated to runtime - BLandNodes ignore ws when creating XMLs. @KavinduZoysa  shall we create and issue for this?

Comment (KavinduZoysa) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLLiteralTest.java:159]: Created an issue https://github.com/ballerina-platform/ballerina-lang/issues/24065

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java (hunk 2 above)
Comment (KavinduZoysa) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java]: We need to use that method, so changed it

**Pull request merged**